src/libraries/System.Text.RegularExpressions/tests/RegexCharacterSetTests.cs

```diff
@@ -33,29 +40,44 @@ public class RegexCharacterSetTests
         [InlineData(@"ACEGIKMOQSUWY", RegexOptions.None, new[] { 'A', 'C', 'E', 'G', 'I', 'K', 'M', 'O', 'Q', 'S', 'U', 'W', 'Y' })]
         [InlineData(@"abcAB", RegexOptions.None, new[] { 'A', 'B', 'a', 'b', 'c' })]
         [InlineData(@"a-c", RegexOptions.None, new[] { 'a', 'b', 'c' })]
+        [InlineData(@"a-fA-F0-9", RegexOptions.None, new[] { 'a', 'b', 'c', 'd', 'e', 'f', 'A', 'B', 'C', 'D', 'E', 'F', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9' })]
         [InlineData(@"X-b", RegexOptions.None, new[] { 'X', 'Y', 'Z', '[', '\\', ']', '^', '_', '`', 'a', 'b' })]
         [InlineData(@"\u0083\u00DE-\u00E1", RegexOptions.None, new[] { '\u0083', '\u00DE', '\u00DF', '\u00E0', '\u00E1' })]
         [InlineData(@"\u007A-\u0083\u00DE-\u00E1", RegexOptions.None, new[] { '\u007A', '\u007B', '\u007C', '\u007D', '\u007E', '\u007F', '\u0080', '\u0081', '\u0082', '\u0083', '\u00DE', '\u00DF', '\u00E0', '\u00E1' })]
         [InlineData(@"\u05D0", RegexOptions.None, new[] { '\u05D0' })]
         [InlineData(@"a\u05D0", RegexOptions.None, new[] { 'a', '\u05D0' })]
         [InlineData(@"\uFFFC-\uFFFF", RegexOptions.None, new[] { '\uFFFC', '\uFFFD', '\uFFFE', '\uFFFF' })]
-        [InlineData(@"a-z-[d-w-[m-o]]", RegexOptions.None, new[] { 'a', 'b', 'c', 'm', 'n', 'n', 'o', 'x', 'y', 'z' })]
+        [InlineData(@"[a-z-[d-w-[m-o]]]", RegexOptions.None, new[] { 'a', 'b', 'c', 'm', 'n', 'n', 'o', 'x', 'y', 'z' })]
         [InlineData(@"\p{IsBasicLatin}-[\x00-\x7F]", RegexOptions.None, new char[0])]
-        [InlineData(@"0-9-[2468]", RegexOptions.None, new[] { '0', '1', '3', '5', '7', '9' })]
+        [InlineData(@"[0-9-[2468]]", RegexOptions.None, new[] { '0', '1', '3', '5', '7', '9' })]
         public void SetInclusionsExpected(string set, RegexOptions options, char[] expectedIncluded)
         {
-            ValidateSet($"[{set}]", options, new HashSet<char>(expectedIncluded), null);
-            if (!set.Contains("["))
+            bool hasBracket = set.Contains("[");
+            if (hasBracket)
             {
+                ValidateSet(set, options, new HashSet<char>(expectedIncluded), null);
+            }
+            else
+            {
+                ValidateSet($"[{set}]", options, new HashSet<char>(expectedIncluded), null);
                 ValidateSet($"[^{set}]", options, null, new HashSet<char>(expectedIncluded));
             }
         }
 
+        [Theory]
+        [InlineData(@"[^1234-[3456]]", RegexOptions.None, new[] { '1', '2', '3', '4', '5', '6' })]
+        public void SetExclusionsExpected(string set, RegexOptions options, char[] expectedExcluded)
+        {
+            ValidateSet(set, options, null, new HashSet<char>(expectedExcluded));
+        }
+
         [Theory]
         [InlineData('\0')]
         [InlineData('\uFFFF')]
         [InlineData('a')]
         [InlineData('5')]
+        [InlineData('\u00FF')]
```

Comment:
Is `\u0100` interesting (verify the 255 boundary)

Reply:
I can add it.  It doesn't currently have any special meaning in the implementation, but that doesn't mean it won't in the future.